pmd-core/src/main/java/net/sourceforge/pmd/cpd/internal/JavaCCTokenizer.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd.internal;
```

Comment:
I didn't heard back from you guys, so I'm raising my concern again.

I don't think this should be internal. We actually want language developers to use this as a starting point. I'm ok with marking it `@Experimental` if we are unsure about the included methods.

Actually, the `AntlrTokenizer` is currently neither internal nor `@Experimental`

Reply:
Some point we discussed on a comment in #1558 is whether implementing third party languages should really be a published API. It's very likely that we're the only ones to work with that API (if we're not then I would assume that the numbers of clients can be counted on one hand). I personally don't think publicly supporting that API is worth the trouble if it's next to unused outside of our codebase. Wdyt?

Reply:
I agree, they are far too few, but there are a couple such as https://github.com/jk1/pmd-kotlin

I fear what kind of message it sends to both them and their users when they can't rely on which versions of PMD it will work, as an internal package means it can be broken at any single point in time.

If I wear them, I would avoid this base class altogether / copy it locally; which kind of misses the point of having a base class in the first place.

Reply:
What you say makes sense too. I've not made up my mind, I think we should discuss that further in the next meeting

Reply:
Let's keep it internal for now. We can make it a public API later, if needed.

*TODO:* Make it consistent with AntlrTokenizer -> this should be marked as internal API then

Reply:
I already did this upon merge